ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReports/OnDemandReportsTab.tsx

```diff
@@ -1,27 +1,127 @@
+    const query: Record<string, string[]> = {
+        'Report State': [],
+    };
+
+    const jobStatusQueryMappings: Record<string, { category: string; value: string }> = {
+        PREPARING: { category: 'Report State', value: 'PREPARING' },
+        WAITING: { category: 'Report State', value: 'WAITING' },
+        ERROR: { category: 'Report State', value: 'FAILURE' },
+        DOWNLOAD_GENERATED: {
+            category: 'Report State',
+            value: 'GENERATED',
+        },
+    };
+
+    const reportJobStatuses = ensureReportJobStatuses(jobStatusFilters);
+
+    reportJobStatuses.forEach((jobStatus) => {
+        const queryMapping = jobStatusQueryMappings[jobStatus];
+        if (queryMapping) {
+            const { category, value } = queryMapping;
+            query[category].push(value);
+        }
+    });
+
+    return query;
+}
+
 function OnDemandReportsTab() {
-    // @TODO: Pass query, pagination, sorting to API call
-    const fetchOnDemandReportsHistoryCallback = useCallback(() => fetchOnDemandReportHistory(), []);
-    const { data, isLoading, error } = useRestQuery(fetchOnDemandReportsHistoryCallback);
+    const { page, perPage, setPage, setPerPage } = useURLPagination(10);
+    const { sortOption, getSortParams } = useURLSort(sortOptions);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const [isViewingOnlyMyJobs, setIsViewingOnlyMyJobs] = useURLStringUnion('viewOnlyMyJobs', [
+        'false',
+        'true',
+    ]);
+
+    const reportJobStatusFilters = ensureStringArray(searchFilter['Report Job Status']);
+
+    const query = getRequestQueryStringForSearchFilter({
+        ...createQueryFromReportJobStatusFilters(reportJobStatusFilters),
+    });
+
+    const fetchOnDemandReportsHistoryCallback = useCallback(
+        () =>
+            fetchOnDemandReportHistory({
+                query,
+                page,
+                perPage,
+                sortOption,
+                showMyHistory: isViewingOnlyMyJobs === 'true',
+            }),
+        [query, page, perPage, sortOption, isViewingOnlyMyJobs]
+    );
+    const { data, isLoading, error, refetch } = useRestQuery(fetchOnDemandReportsHistoryCallback, {
+        clearErrorBeforeRequest: false,
+    });
 
     // @TODO: Add polling
 
     const tableState = getTableUIState({
         isLoading,
         data,
         error,
-        searchFilter: {},
-        isPolling: false,
+        searchFilter,
+        isPolling: true,
     });
 
+    const onReportJobStatusFilterChange = (_checked: boolean, selectedStatus: ReportJobStatus) => {
+        const isStatusIncluded = reportJobStatusFilters.includes(selectedStatus);
+        const newFilters = isStatusIncluded
+            ? ensureReportJobStatuses(
+                  reportJobStatusFilters.filter((status) => status !== selectedStatus)
+              )
+            : ensureReportJobStatuses([...reportJobStatusFilters, selectedStatus]);
```

Comment:
Nit: I've implemented similar logic myself a handful of times. It might be nice if we had an `addOrRemoveItem()` (or other naming?) utlity that handles the case of an item being toggled in an array.

Reply:
Good idea. I'll try to find where else we do this and see if I can abstract the logic out.

Reply:
I added a function called `toggleItemInArray` and placed it in the currently existing `utils/arrayUtils` file.